test/physics/em/TsaiUrbanDistribution.test.cc

```diff
@@ -0,0 +1,54 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file TsaiUrbanDistribution.test.cc
+//---------------------------------------------------------------------------//
+#include "physics/em/detail/TsaiUrbanDistribution.hh"
+
+#include <random>
+#include "base/Constants.hh"
+#include "base/Units.hh"
+#include "celeritas_test.hh"
+
+using celeritas::detail::TsaiUrbanDistribution;
+
+//---------------------------------------------------------------------------//
+// TEST HARNESS
+//---------------------------------------------------------------------------//
+
+class TsaiUrbanDistributionTest : public celeritas::Test
+{
+  protected:
+    void SetUp() override {}
+
+    std::mt19937 rng;
+};
+
+//---------------------------------------------------------------------------//
+// TESTS
+//---------------------------------------------------------------------------//
+
+TEST_F(TsaiUrbanDistributionTest, bin)
+{
+    using namespace celeritas::constants;
+    using namespace celeritas::units;
+
+    std::vector<float> angles;
+    // Loop over various electron energies (converted to MeVEnergy)
+    for (double inc_e : {1.0, 10.0, 20.0, 50.0, 100.0})
+    {
+        TsaiUrbanDistribution sample_angle(MevEnergy{inc_e},
+                                           MevMass{electron_mass});
+        double                angle = sample_angle(rng);
+        angles.push_back(angle);
+    }
+
+    const float expected_angles[] = {0.3245577893213,
+                                     -0.02649751911085,
+                                     0.6002972458021,
+                                     0.9822511633688,
+                                     0.1573926944923};
+    EXPECT_VEC_EQ(expected_angles, angles);
```

Comment:
Use `EXPECT_VEC_SOFT_EQ` for floating point "equivalence" (with a bit of tolerance built in) -- is this why you chose to use a vector of floats instead of doubles?

Reply:
Yes, `double` was giving erroneous results even when the values were the same to the precision shown in the output.

Reply:
`EXPECT_VEC_SOFT_EQ` is fine for `double` "equivalence" -- thanks!